### PR TITLE
update system-contracts.md to bedrock version

### PR DESCRIPTION
**Pull request by opfocus**
**Description**

Fix partial broken links, and replace the installed package with @eth-optimism/contracts-bedrock:0.15.0

**Tests**
```
E:\standard-bridge-standard-token>npm list
standard-bridge-standard-token@1.0.0 E:\standard-bridge-standard-token
+-- @eth-optimism/contracts-bedrock@0.15.0
+-- @eth-optimism/sdk@3.1.0
+-- @nomiclabs/hardhat-ethers@2.2.3
+-- dotenv@16.3.1
`-- hardhat@2.17.1


E:\standard-bridge-standard-token>npx hardhat console --network goerli
Welcome to Node.js v18.12.1.
Type ".help" for more information.
> const {getContractInterface, getContractDefinition, predeploys} = require("@eth-optimism/contracts-bedrock")
undefined
> const artifact = getContractDefinition('L1CrossDomainMessenger')
undefined
> const abi = artifact.abi
undefined
> const bytecode = artifact.bytecode
> abi
[
  {
    inputs: [ [Object] ],
    stateMutability: 'nonpayable',
    type: 'constructor'
  },
.....

> abi
[
  {
    inputs: [ [Object] ],
    stateMutability: 'nonpayable',
    type: 'constructor'
  },

```

**Additional context**

Add any other context about the problem you're solving.

**Metadata**

- Fixes #[Link to Issue]

### CODE REVIEW

File: src/docs/developers/build/system-contracts.md [hunk 1]
```diff
@@ -27,52 +27,52 @@ Now you simply need to import the appropriate contracts.
 
 ### Installing via NPM or Yarn
 
-We export a package [`@eth-optimism/contracts`](https://www.npmjs.com/package/@eth-optimism/contracts?activeTab=readme) that makes it easy to use the OP Mainnet contracts within NPM or Yarn based projects.
+We export a package [`@eth-optimism/contracts-bedrock`](https://www.npmjs.com/package/@eth-optimism/contracts-bedrock) that makes it easy to use the OP Mainnet contracts within NPM or Yarn based projects.
 Install the package as follows:
 
 ```
-npm install @eth-optimism/contracts
+npm install @eth-optimism/contracts-bedrock@0.15.0
```
Comment (sbvegan) [inline src/docs/developers/build/system-contracts.md:34]: lets remove the version:

`npm install @eth-optimism/contracts-bedrock`

File: src/docs/developers/build/system-contracts.md [hunk 2]
```diff
@@ -27,52 +27,52 @@ Now you simply need to import the appropriate contracts.
...
 
 ### Installing via NPM or Yarn
 
-You can use the [`@eth-optimism/contracts`](https://www.npmjs.com/package/@eth-optimism/contracts?activeTab=readme) package to interact with the OP Mainnet system contracts from a JavaScript or TypeScript based project.
+You can use the [`@eth-optimism/contracts-bedrock`](https://www.npmjs.com/package/@eth-optimism/contracts-bedrock) package to interact with the OP Mainnet system contracts from a JavaScript or TypeScript based project.
 Install the package as follows:
 
 ```
-npm install @eth-optimism/contracts
+npm install @eth-optimism/contracts-bedrock@0.15.0
```
Comment (sbvegan) [inline src/docs/developers/build/system-contracts.md:67]: lets remove the version:

`npm install @eth-optimism/contracts-bedrock`

File: src/docs/developers/build/system-contracts.md [hunk 3]
```diff
@@ -27,52 +27,52 @@ Now you simply need to import the appropriate contracts.
...
 ```
 
 ### Getting contract artifacts, interfaces, and ABIs
 
 You can get the compiler artifact, bytecode, and ABI for any OP Mainnet contract as follows:
 
 ```ts
-import { getContractDefinition } from '@eth-optimism/contracts'
+import { getContractDefinition } from '@eth-optimism/contracts-bedrock'
```
Comment (sbvegan) [inline src/docs/developers/build/system-contracts.md:75]: So installed the new package and it looks like this isn't valid anymore. It needs to be updated. It looks like `@eth-optimism/contracts-bedrock` uses foundry, so we'll have to adjust this to reflect that update.

File: src/docs/developers/build/system-contracts.md [hunk 4]
```diff
@@ -83,7 +83,7 @@ const deployedBytecode = artifact.deployedBytecode
 Similarly, you can also get [ethers Interface objects](https://docs.ethers.io/v5/api/utils/abi/interface/) for any contract:
 
 ```ts
-import { getContractInterface } from '@eth-optimism/contracts'
+import { getContractInterface } from '@eth-optimism/contracts-bedrock'
```
Comment (sbvegan) [inline src/docs/developers/build/system-contracts.md:86]: same as the comment above

File: src/docs/developers/build/system-contracts.md [hunk 5]
```diff
@@ -93,7 +93,7 @@ const iface = getContractInterface('SomeOptimismContract')
 You can get the address of any L2 contract as follows:
 
 ```ts
-import { predeploys } from '@eth-optimism/contracts'
+import { predeploys } from '@eth-optimism/contracts-bedrock'
```
Comment (sbvegan) [inline src/docs/developers/build/system-contracts.md:96]: same as the comment above

**Review by sbvegan — COMMENTED** [review]
Thank you for catching this outdated guide, we still need to make some updates to reflect the new package. Please let me know if you're up to making those changes.

File: src/docs/developers/build/system-contracts.md (hunk 1 above)
Comment (opfocus) [reply src/docs/developers/build/system-contracts.md:34]: I attempted to update @eth-optimism/contracts-bedrock to the latest version, but I found that it couldn't directly achieve the functionality described in the documentation. I made compromises in this regard, and I am currently monitoring updates to the package or whether there are any usage methods with the new package that I haven't yet grasped.

File: src/docs/developers/build/system-contracts.md (hunk 3 above)
Comment (opfocus) [reply src/docs/developers/build/system-contracts.md:75]: > So installed the new package and it looks like this isn't valid anymore. It needs to be updated. It looks like `@eth-optimism/contracts-bedrock` uses foundry, so we'll have to adjust this to reflect that update.


Noted, I will take another look at it under guidance.

Comment (opfocus) [reply src/docs/developers/build/system-contracts.md:75]: > So installed the new package and it looks like this isn't valid anymore. It needs to be updated. It looks like `@eth-optimism/contracts-bedrock` uses foundry, so we'll have to adjust this to reflect that update.
I've encountered some difficulties in retrieving ABI information using Foundry. As a beginner, I might not be able to accomplish it effectively. Therefore, I am willing to have this task completed by anyone. Until that happens, I will continue preparing for my next submission related to this content.

Comment (opfocus) [reply src/docs/developers/build/system-contracts.md:75]: > So installed the new package and it looks like this isn't valid anymore. It needs to be updated. It looks like `@eth-optimism/contracts-bedrock` uses foundry, so we'll have to adjust this to reflect that update.

I've encountered some difficulties in retrieving ABI information using Foundry. As a beginner, I might not be able to accomplish it effectively. Therefore, I am willing to have this task completed by anyone. Until that happens, I will continue preparing for my next submission related to this content.

Comment (sbvegan) [reply src/docs/developers/build/system-contracts.md:75]: Thank you for your work here. When I get the time, I'll try and bring it over the finish line.

**Comment by opfocus** [comment]
In the document's section titled "#Interacting from the client side," the latest `contracts-bedrock` package seems to no longer support this portion of the content. The` contracts-ts` package appears to achieve the same objective. I conducted some testing on the contracts package, and the following commands are viable: 
```
const { optimismMintableErc20FactoryABI, optimismMintableErc20FactoryAddress } = require('@eth-optimism/contracts-ts');
optimismMintableERC20Factory = new ethers.Contract(
  optimismMintableErc20FactoryAddress[1],
  optimismMintableErc20FactoryABI,
  await ethers.getSigner()
);
```

File: src/docs/developers/build/system-contracts.md (hunk 2 above)
Comment (opfocus) [reply src/docs/developers/build/system-contracts.md]: Has been removed

File: src/docs/developers/build/system-contracts.md (hunk 1 above)
Comment (opfocus) [reply src/docs/developers/build/system-contracts.md]: Has been removed

File: src/docs/developers/build/system-contracts.md (hunk 3 above)
Comment (opfocus) [reply src/docs/developers/build/system-contracts.md]: "@eth-optimism/contracts-bedrock" has been replaced with "@eth-optimism/contracts-ts" in this section of the content. Below, I will provide a test and a summary of my thoughts.

File: src/docs/developers/build/system-contracts.md (hunk 4 above)
Comment (opfocus) [reply src/docs/developers/build/system-contracts.md]: I have removed this section of content. One reason for the removal is that I am not very familiar with the purpose of interfaces. Another reason is that it seems interfaces can be conveniently generated through ABI using ethers.utils.Interface(ABI).

File: src/docs/developers/build/system-contracts.md (hunk 5 above)
Comment (opfocus) [reply src/docs/developers/build/system-contracts.md]: "I have deleted this section of content. The reason for the deletion is that in the previous content, we have already obtained the system's address constants."

**Comment by opfocus** [comment]
Replacing '@eth-optimism/contracts-bedrock' with '@eth-optimism/contracts-ts' was a compromise. It appears that the latest 'contracts-bedrock' does not provide certain APIs, but these data can still be obtained through reading and parsing JSON files. When I tried the new 'contracts-ts' package, I found that it offers these APIs, so I included it. Overall, they worked together to accomplish the tasks outlined in the previous documentation.

**Comment by opfocus** [comment]
I'm learning and sharing. There's still a long way to go before making contributions. I'm working hard, and I would greatly appreciate any advice you can provide.

**Comment by Nicca42** [comment]
The developer docs have been moved out of this repo.

**Pull request closed without merging**